src/uct/ib/rc/accel/rc_mlx5_ep.c

```diff
@@ -559,13 +565,47 @@ void uct_rc_mlx5_ep_vfs_populate(uct_rc_ep_t *rc_ep)
     uct_rc_txqp_vfs_populate(&ep->super.txqp, ep);
 }
 
+static ucs_status_t
+uct_rc_mlx5_ep_flush_remote(uct_ep_h tl_ep, uct_completion_t *comp)
+{
+    const unsigned get_length = 8;
+    UCT_RC_MLX5_EP_DECL(tl_ep, iface, ep);
+    uct_rc_iface_send_desc_t *desc;
+
+    UCT_RC_CHECK_RES(&iface->super, &ep->super);
+
+    UCT_RC_IFACE_GET_TX_DESC(iface, &iface->super.tx.mp, desc);
+    desc->super.handler   = uct_rc_ep_flush_remote_handler;
+    desc->super.user_comp = comp;
+
+    uct_rc_mlx5_common_txqp_bcopy_post(iface, IBV_QPT_RC, &ep->super.txqp,
+                                       &ep->tx.wq, MLX5_OPCODE_RDMA_READ,
+                                       get_length, 0,
+                                       ep->super.flush_rkey, NULL, NULL,
+                                       0, MLX5_WQE_CTRL_CQ_UPDATE, 0, desc,
+                                       desc + 1, NULL);
+    ep->super.flags &= ~UCT_RC_EP_FLAG_FLUSH_REMOTE;
+
+    return UCS_INPROGRESS;
+}
+
 ucs_status_t uct_rc_mlx5_ep_flush(uct_ep_h tl_ep, unsigned flags,
                                   uct_completion_t *comp)
 {
     UCT_RC_MLX5_EP_DECL(tl_ep, iface, ep);
     int already_canceled = ep->super.flags & UCT_RC_EP_FLAG_FLUSH_CANCEL;
     ucs_status_t status;
 
+    UCT_CHECK_PARAM(!ucs_test_all_flags(flags, UCT_FLUSH_FLAG_CANCEL |
+                                               UCT_FLUSH_FLAG_REMOTE),
+                    "flush flags CANCEL and REMOTE are mutually exclusive");
+
+    if ((flags & UCT_FLUSH_FLAG_REMOTE) &&
+        (ep->super.flags & UCT_RC_EP_FLAG_FLUSH_REMOTE) &&
+        !(ep->super.flush_rkey & 0xff)) {
```

Comment:
maybe better define some inline function for checking rkey validity?
probably somehow common with `uct_ib_md_is_flush_rkey`

Reply:
yep, good idea.
done